build/tools.mk

```diff
@@ -11,8 +11,9 @@ BUILD_DIR        ?= ./bin/
 # Go file to track tool deps with go modules
 TOOL_DIR     ?= tools
 TOOL_CONFIG  ?= $(TOOL_DIR)/tools.go
+DEPENDENCIES_FILE ?= dependencies.txt
 
-GOTOOLS ?= $(shell cd $(TOOL_DIR) && go list -f '{{ .Imports }}' -tags tools |tr -d '[]')
+GOTOOLS ?= $(shell cd $(TOOL_DIR) && cat $(DEPENDENCIES_FILE))
```

Comment:
#### 1

##### Context
As described earlier, in order to upgrade `nfpm` and `goreleaser` to fix a current dependabot alert ([#26](https://github.com/newrelic/terraform-provider-newrelic/security/dependabot/26)), `go` needed to be upgraded to `1.21` as trying to upgrade these modules on `1.19` threw the error
```
imported by a module that requires go1.21
```
and hence, the effort to upgrade `go` to `1.21`. 

Upon trying to upgrade to `go1.21` with the existing structure of the provider, [the following error](https://github.com/newrelic/terraform-provider-newrelic/actions/runs/7511051913/job/20450195022?pr=2540) appeared to be thrown

```
package github.com/newrelic/terraform-provider-newrelic/v2/tools: build constraints exclude all Go files in /home/runner/work/terraform-provider-newrelic/terraform-provider-newrelic/tools
```

which indicated a probable issue with build constraints in `tools.go` in the tools directory. Upon removing these build constraints for testing locally, a new error was locally encountered

```
import XXX a program, not an importable package
```

##### Workaround
This is how we've been able to overcome the `build constraints exclude all Go files in /home/runner/work/terraform-provider-newrelic/terraform-provider-newrelic/tools` error - placing the list of dependencies in a different source and running a `go list` on the text file (changing the command used in `tools.mk`). With this, the compilation failure seems to have disappeared.